src/hooks/useBalance.ts

```diff
@@ -45,29 +45,31 @@ const useBalance = (api: ApiPromise, address: string, providedSi: boolean = fals
   const [state, setState] = useMountedState<State>(initValues);
 
   useEffect((): (() => void) => {
-    let unsubscribe: null | (() => void) = null;
-    api?.query?.system
-      .account(address, ({ data }): void => {
-        setState({
-          chainTokens: data.registry.chainTokens[0],
-          formattedBalance: formatBalance(data.free, {
-            decimals: api.registry.chainDecimals[0],
-            withUnit: api.registry.chainTokens[0],
-            withSi: providedSi
-          }),
-          free: data.free
+    const getBalance = async (api: ApiPromise, address: string, setState: any): Promise<VoidFn> => {
+      try {
+        const u = await api.query.system.account(address, ({ data }): void => {
+          setState({
+            chainTokens: data.free.registry.chainTokens[0],
+            formattedBalance: formatBalance(data.free, {
+              decimals: api.registry.chainDecimals[0],
+              forceUnit: '-',
+              withSi: providedSi
```

Comment:
Ugh, somehow that comment did not get added earlier:

Please revert the `withUnit` stuff, cause I fixed it way too many times already.